bundles/org.openhab.voice.rustpotterks/src/main/resources/OH-INF/config/config.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<config-description:config-descriptions
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:config-description="https://openhab.org/schemas/config-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/config-description/v1.0.0
+		https://openhab.org/schemas/config-description-1.0.0.xsd">
+	<config-description uri="voice:rustpotterks">
+		<parameter-group name="wakewordDetector">
+			<label>Wakeword Detector</label>
+			<description>Wakeword detection options.</description>
+		</parameter-group>
+		<parameter-group name="vadDetector">
+			<label>VAD Detector</label>
+			<description>Optional voice activity detector options.</description>
+		</parameter-group>
+		<parameter name="threshold" type="decimal" min="0" max="1" groupName="wakewordDetector">
+			<label>Threshold</label>
+			<description>Configures the detector threshold, is the min score (in range 0. to 1.) that some of the wakeword
+				templates should obtain to trigger a detection. Model defined value takes prevalence if present.</description>
+			<default>0.5</default>
+		</parameter>
+		<parameter name="averagedThreshold" type="decimal" min="0" max="1" groupName="wakewordDetector">
+			<label>Averaged Threshold</label>
+			<description>Configures the detector averaged threshold, is the min score (in range 0. to 1.) that the averaged
+				wakeword template should obtain to allow to continue with the detection. This way it can prevent to run the
+				comparison of the current frame against each of the wakeword templates. If set to 0 this functionality is disabled.</description>
```

Comment:
Yes, it's maybe not very understandable.

When generating a model, the model will contain internally one set of features for each wav sample that you use to generate it, but will also contains and extra set of features that is generated by averaging the other sets.

When configuring the averaged threshold, you are configuring the minimum score that the current frame should obtain against this extra set of features to continue with the detection process (which is scoring against each of the other feature sets).

You will get low scores agains the averaged feature set and more false positives, but with the correct value you will also avoid running the whole detection process in most of the audio frames (you will be running the comparison 1 time instead of one per sample). This reduces a lot the cpu usage when you are talking or watching tv near the detector, as the comparison step is the more cpu expensive step.

Reply:
Do you know how can I improve the comment to summarize that?

Reply:
I will add that info to the rustpotter readme.

Reply:
Ok, I think I understand better even if the notion of set of features is not yet clear for me.
Maybe you could describe your 2 thresholds as detection threshold for main set of features and detection threshold for extra set of features ?

Reply:
Yeah what I mean by a set of features, is an arrays of arrays (ie matrix), that the library generates from the wav sample. Each member in the array is an array of 44 numbers that is generated from each 100ms of audio called its feature spectrum.

Reply:
> Maybe you could describe your 2 thresholds as detection threshold for main set of features and detection threshold for extra set of features ?
Yeah but I think it's important to highlight that this extra "feature set" has been generated by combining the others.

Reply:
@lolodomo I think this is the last one, could this be a better description?:
```
Configures the detector averaged threshold, is the min score (in range 0. to 1.) that the audio should obtain agains a combination of the wake word templates, the detection will be aborted if this is not the case. This way it can prevent to run the comparison of the current frame against each of the wake word templates which saves cpu. If set to 0 this functionality is disabled.
```

Reply:
Yes, your proposal looks good to me, this is now clear what is this setting.